src/ios/AppDelegate+notification.m

```diff
@@ -63,11 +63,42 @@ - (AppDelegate *)pushPluginSwizzledInit
                                                 name:UIApplicationDidBecomeActiveNotification
                                               object:nil];
 
+    // Cordova Plugin Push issue #1. Define a listener for the UIApplicationDidFinishLaunchingNotification event that will be invoked when the app has  
+    // finished launching and ready to present any windows to the user (according to the lifecycle of the iOS app).
+    // https://github.com/TransformativeMed/cordova-plugin-push/issues/1
+    // Helpful documentation about lifecycle: https://medium.com/@theiOSzone/briefly-about-the-ios-application-lifecycle-92f0c830b754
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(getDataNotificationLaunchedApp:)
+                                                 name:UIApplicationDidFinishLaunchingNotification
+                                               object:nil];
+
     // This actually calls the original init method over in AppDelegate. Equivilent to calling super
     // on an overrided method, this is not recursive, although it appears that way. neat huh?
     return [self pushPluginSwizzledInit];
 }
 
+// Cordova Plugin Push issues #1. This code will be called immediately after UIApplicationDidFinishLaunchingNotification event once the app is opened/loaded correctly.
+// https://github.com/TransformativeMed/cordova-plugin-push/issues/1
```

Comment:
Let's explain the issue and the context here but let's not refer to TransformativeMed-specific issues. We will likely try to open a pull request to get these changes back into the parent repo. They wouldn't want to see URLs in the comments to private issues that they can't open.

Reply:
sure, let me remove the URLs that I added in all comments. But, I have a question here:

Since this behavior is a custom feature that was requested by the solution delivery team time ago, should we propose these changes to the plugin creator? 

The reason why I'm asking this is: 

the plugin already has the same logic to check permissions in the `init` method, but in our app that method will be invoked once the **site code registration** and **login** have been successfully completed (since we require the session values to send the `device_id` to our CORES API). Solution Delivery team suggested time ago to always request the permissions when the app starts, no matter if there is no site code, that's why I added that behavior to the push plugin.

What do you think?